injectproxy/enforce.go

```diff
@@ -21,7 +21,7 @@ import (
 )
 
 type Enforcer struct {
-	labelMatchers map[string]*labels.Matcher
```

Comment:
Why we need access to this? It looks like code smell, as it leaks the enforcer internal logic. Aren't its methods enough? Are you missing some functionality that could be added as method?

Reply:
We need to access this because enforceMatchers is a method of enforcer and to access it first we need to create an enforcer(in our use case in Thanos this enforcer contains the information about the tenant label name we want to replace and tenant access a user has as specified in the header)

Reply:
yea, but this method has access to this because it's a method on `Enforcer` itself, no?

Reply:
hmm, I think we can use [NewEnforcer](https://github.com/prometheus-community/prom-label-proxy/blob/38b1382d7368e994e39c0cb853f3f3c9300f5045/injectproxy/enforce.go#L27) to create a new enforcer.
Let me try it.

Reply:
Well. I think we even have to! That's why we do constructors in Go (:

Reply:
ahh just refreshed those concepts thanks! definitely make sense.